scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/StaxJobFactory.java

```diff
@@ -1495,12 +1523,14 @@ private void makeDependences() throws JobCreationException {
      * @return the string with variables replaced by values.
      * @throws JobCreationException if a Variable has not been found
      */
-    private String replace(String str) throws JobCreationException {
+    private String replace(String str, Map<String, String> variables) throws JobCreationException {
         Map<String, String> replacements = new HashMap<>();
-        for (Map.Entry o : System.getProperties().entrySet()) {
+        for (Map.Entry<Object, Object> o : System.getProperties().entrySet()) {
```

Comment:
I was not aware of this replacement :-). The documentation does not mention it. In general, variables documentation should be improved.